docs/src/simulation.md

```diff
@@ -0,0 +1,183 @@
+Simulating an Experiment from Scratch
+======================================
+
+Here is a worked example of simulating a partially crossed design from scratch.
+
+First, some setup:
+
+```@example Main
+using DataFrames
+using LinearAlgebra
+using MixedModels, MixedModelsSim
+using Random
+```
+
+## Assemble the Design
+
+We're going to do a 2 x 2 x 2 design.
+For concreteness, let's think of this as a linguistic design:
+
+- **age** `old` vs. `young`, _between subjects_
+- **frequency** `high` vs. `low`, _between items_
+- **context** `matched` vs. `unmatched`, _within both_.
+
+Further, let's assume we want 40 subjects and 80 items.
+We can specify this design as follows:
+
+```@example Main
+n_subj = 40
+n_item = 80
+subj_btwn = Dict(:age => ["old", "young"])
+item_btwn = Dict(:frequency => ["high", "low"])
+both_win = Dict(:context => ["matched", "unmatched"])
+```
+
+and then generate it:
+
+```@example Main
+rng = MersenneTwister(42)  # specify our random number generator for reproducibility
+design = simdat_crossed(rng, n_subj, n_item;
+                        subj_btwn = subj_btwn,
+                        item_btwn = item_btwn,
+                        both_win = both_win)
+```
+
+Note that `simdat_crossed` returns a row table, which `MixedModels.jl` can process directly.
+For nicely displaying it, we can also convert it to a dataframe and change the factors to use pooled arrays to save a bit of memory.
+
+```@example Main
+df = pooled!(DataFrame(design))
+first(df, 5)
+```
+
+Note that `simdat_crossed` generated a column `dv` for our dependent variable that has been pre-populated with noise from a standard normal distribution ($N(0,1)$).
+Typically, we will want to scale that, but we can do that in the simulation step.
+Also, this dependent variable is *pure noise*: we haven't yet added in effects.
+Adding in effects also comes in the simulation step.
+
+But before we get to simulating, let's fit the model to the noise, just to see how things look. We're going to use effects coding for our contrasts.
+
+```@example Main
+contrasts = Dict(:age => EffectsCoding(base="young"),
+                 :frequency => EffectsCoding(base="high"),
+                 :context => EffectsCoding(base="matched"))
+form = @formula(dv ~ 1 + age * frequency * context +
+                    (1 + frequency + context | subj) +
+                    (1 + age + context | item))
+m0 = fit(MixedModel, form, design; contrasts=contrasts)
+```
+
+## Assemble the Random Effects
+
+The hard part in simulating right now is specifying the random effects.
+We're working on making this bit easier, but you need to specify the variance-covariance matrix of the random effects. You can see what this
+looks like:
+
+```@example Main
+vc = VarCorr(m0)
+```
+
+```@example Main
+vc.σρ
+```
+
+In matrix form:
+
+```@example Main
+m0.λ
+```
+
+```@example Main
+m0.λ[1]
+```
+
+```@example Main
+m0.λ[2]
+```
+
```

Comment:
I predict this calculation will be a stumbling block for many.  Would it be worthwhile allowing specification of the theta parameters from standard deviations and correlations as an alternative?

Reply:
Yes, I completely agree. 

The `flatlowertri` and `update` function should help a bit, but maybe it makes sense to expand that a bit and create a 
`update!(::MixedModel, Varargs{Matrix})` method. (So that the user doesn't even have to load `LinearAlgebra` and use `LowerTriangular`).

Reply:
Actually, how about something like:

```julia
function create_re(sigmas....; corrmat=Matrix{Float64}(I, length(sigmas), length(sigmas))
    ss = diagm(sigmas)
    cholesky(diagm(ss) * corrmat * diagm(ss)).L
end

update!(m::LinearMixedModel, re...) = update!(m, vcat(flatlowertri.(re...)...))

subjre = create_re(vv1, vv2, vv3)
itemre = create_re(vv4 vv5, vv6)

update!(m, subjre, itemre)
```

Reply:
Thank you for adding that.  It looks fine now.